src/components/client/campaign-news/CampaignNewsList.tsx

```diff
@@ -0,0 +1,182 @@
+  },
+
+  [`& .${classes.articleDescription}`]: {
+    fontSize: theme.typography.pxToRem(16),
+    fontFamily: theme.typography.fontFamily,
+  },
+
+  [`& .${classes.dateAndAuthorContainer}`]: {
+    marginBottom: theme.spacing(2),
+  },
+
+  [`& .${classes.readMoreButton}`]: {
+    fontSize: theme.typography.pxToRem(14),
+    color: theme.palette.primary.light,
+    textDecoration: 'underline',
+    padding: 0,
+    margin: 0,
+  },
+
+  [`div.${classes.articleDescription} > p`]: {
+    margin: 0,
+  },
+}))
+
+type Props = {
+  articles: CampaignNewsResponse[] | []
+}
+
+export default function CampaignNewsList({ articles }: Props) {
+  const { t, i18n } = useTranslation('news')
+  const CHARACTER_LIMIT = 350
+  const [isExpanded, expandContent] = useShowMoreContent()
+
+  return (
+    <>
+      {articles?.map((article, index) => {
+        const documents = GetArticleDocuments(article.newsFiles)
+        const images = GetArticleGalleryPhotos(article.newsFiles)
+        const sanitizedDescription = sanitizeHTML(article.description)
+        return (
+          <Grid
+            container
+            key={article.id}
+            sx={{
+              backgroundColor: index % 2 === 0 ? '#FFFFFF' : '#E3E3E3',
+              borderBottom: 1,
+              borderColor: index % 2 === 0 ? '#FFFFFF' : '#C4C4C4',
+            }}>
+            <ArticleSection>
+              <Grid container columnGap={2} rowGap={1} className={classes.dateAndAuthorContainer}>
+                <Grid container item gap={1} xs="auto">
+                  <AvTimerIcon color="primary" />
+                  <Typography className={classes.articlepublishedDate}>
+                    {formatDateString(article.publishedAt, i18n.language)} &nbsp;
+                    {dateToTime(article.publishedAt, i18n.language)}
+                  </Typography>
+                </Grid>
+                <Grid container item gap={1} xs="auto" style={{ maxWidth: '100%' }} wrap="nowrap">
+                  <SupervisedUserCircleOutlinedIcon color="primary" />
+                  <Typography className={classes.articleAuthor}>{article.author}</Typography>
+                </Grid>
+              </Grid>
+              <Grid container rowGap={1} columnGap={4}>
+                <Grid
+                  container
+                  item
+                  direction={'column'}
+                  gap={1}
+                  xs={'auto'}
+                  style={{ maxWidth: '100%' }}>
+                  <Typography className={classes.articleHeader}>{article.title}</Typography>
+                  <Grid container item>
+                    {!isExpanded[article.id] && sanitizedDescription.length > CHARACTER_LIMIT ? (
+                      <Typography
+                        component={'div'}
+                        className={classes.articleDescription}
+                        dangerouslySetInnerHTML={{
+                          __html: sanitizedDescription.slice(0, CHARACTER_LIMIT) + '...',
```

Comment:
I assume you needed to set the html from Quill via dangerouslySetInnerHTML so that you can limit the length of the description. 
A safer would be to do with the Quill editor in readonly mode but I don't know how to limit the text size, so unitl we find how, this way is ok.

Reply:
The quill editor seems to use innerHTML under the hood to display the HTML contents, which is the exact same thing as  setting `dangerouslySetInnerHTML` in react. Not quite sure if it is any safer, in regards of outputting HTML, but the HTML content is being sanitized anyway.

Using `dangerouslySetInnerHTML` also should be tad a bit faster, as the HTML content is displayed directly, rather than having to wait for the React Quill editor to load.